textdb/textdb-common/src/main/java/edu/uci/ics/textdb/common/field/Span.java

```diff
@@ -16,4 +40,36 @@ public int getStart() {
     public int getEnd() {
         return end;
     }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + end;
+        result = prime * result
+                + ((fieldName == null) ? 0 : fieldName.hashCode());
+        result = prime * result + start;
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        Span other = (Span) obj;
+        if (end != other.end)
+            return false;
+        if (fieldName == null) {
+            if (other.fieldName != null)
+                return false;
+        } else if (!fieldName.equals(other.fieldName))
+            return false;
+        if (start != other.start)
+            return false;
+        return true;

```

Comment:
Correct.  In general, we could mark the same substring with different labels (e.g., keys).  Our test cases rely on a strict equality comparison.

Please follow the order of the member variables to do the comparisons.